src/nbl/asset/utils/CHLSLCompiler.cpp

```diff
@@ -25,22 +25,712 @@
 #include <boost/wave/cpplexer/cpp_lex_token.hpp>
 #include <boost/wave/cpplexer/cpp_lex_iterator.hpp>
 
-#define TCPP_IMPLEMENTATION
-#include <tcpp/source/tcppLibrary.hpp>
-#undef TCPP_IMPLEMENTATION
-
 using namespace nbl;
 using namespace nbl::asset;
 using Microsoft::WRL::ComPtr;
 
 static constexpr const wchar_t* SHADER_MODEL_PROFILE = L"XX_6_7";
 
-namespace nbl::asset::hlsl::impl
+
+namespace nbl::wave {
+    using namespace boost;
+    using namespace boost::wave;
+    using namespace boost::wave::util;
+
+
+    class include_paths
+    {
+    private:
+        typedef std::list<std::pair<system::path, std::string> >
+            include_list_type;
+        typedef include_list_type::value_type include_value_type;
+
```

Comment:
if you're gonna slap that much "foreign" code in, then make a new header in `src`

Reply:
Also can you please ANNOTATE `// Nabla change start` `//Nabla change end` what you changed from boost ?